File: wrappers/java/src/test/java/org/hyperledger/indy/sdk/payment/BuildSetTxnFeesRequestTest.java
package org.hyperledger.indy.sdk.payment;

import org.hyperledger.indy.sdk.InvalidStructureException;
import org.hyperledger.indy.sdk.payments.Payments;
import org.hyperledger.indy.sdk.payments.UnknownPaymentMethodException;
import org.junit.Test;

import java.util.concurrent.ExecutionException;

import static org.hamcrest.CoreMatchers.isA;

public class BuildSetTxnFeesRequestTest extends PaymentIntegrationTest {

	@Test
	public void testBuildSetTxnFeesRequestWorksForUnknownPaymentMethod() throws Exception {
		thrown.expect(ExecutionException.class);
		thrown.expectCause(isA(UnknownPaymentMethodException.class));

		Payments.buildSetTxnFeesRequest(wallet, DID_TRUSTEE, paymentMethod, fees).get();
	}

	@Test
	public void testBuildSetTxnFeesRequestWorksForInvalidFees() throws Exception {
		thrown.expect(ExecutionException.class);
		thrown.expectCause(isA(InvalidStructureException.class));

		Payments.buildSetTxnFeesRequest(wallet, DID_TRUSTEE, paymentMethod, "[txnType1:1, txnType2:2]").get();
	}
}

File: wrappers/dotnet/indy-sdk-dotnet-test/PaymentTests/BuildSetTxnFeesRequestTest.cs
using Hyperledger.Indy.PaymentsApi;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System.Threading.Tasks;

namespace Hyperledger.Indy.Test.PairwiseTests
{
    [TestClass]
    public class BuildSetTxnFeesRequestTest : PaymentIntegrationTestBase
    {
        [TestMethod]
        public async Task TestBuildSetTxnFeesRequestWorksForUnknownPaymentMethod()
        {
            var ex = await Assert.ThrowsExceptionAsync<UnknownPaymentMethodException>(() =>
                 Payments.BuildSetTxnFeesRequestAsync(wallet, DID_TRUSTEE, paymentMethod, fees)
             );
        }

        [TestMethod]
        public async Task testBuildSetTxnFeesRequestWorksForInvalidFees()
        {
            var ex = await Assert.ThrowsExceptionAsync<InvalidStructureException>(() =>
                 Payments.BuildSetTxnFeesRequestAsync(wallet, DID_TRUSTEE, paymentMethod, "[txnType1:1, txnType2:2]")
             );
        }
    }
}
